### DISCUSSION TITLE
Independent PowerDNS Authoritative Primary and two Authoritative Server Secondary

### DISCUSSION THREAD

**Discussion by sysbitnet** [PowerDNS/pdns#14532] [Q&A]
I read too many pages and documents on the Internet, but I never found how to configure independent servers to work stably.

For that reason, I started to write documents, and when I am finished, I plan to publish and share them so I can help someone with school, and jobs...

In PowerDNS Authoritative - read official [documentation](https://docs.powerdns.com/authoritative/settings.html) 

My goal is to implement a PowerDNS Authoritative server architecture with a single Primary server and two Secondary servers. This setup is designed for both security and efficiency.

The PowerDNS Authoritative Primary server will act as the central management hub. It will be an internal-only server, completely isolated from public client queries. Its sole purpose is to:

Store and manage all DNS zones (adding, changing, deleting records, or deleting zone).

**Automatically notify the Secondary servers whenever a zone is updated.**

The two PowerDNS Authoritative Secondary servers will be the only public-facing servers. They will:

Receive all client DNS queries from the internet.

Receive all zone updates from the Primary server.

Serve all DNS responses to clients.

This architecture ensures that the source of truth—the Primary server—is protected from external access, while the Secondary servers handle all of the public DNS traffic and provide redundancy.

For Example 
<img width="1024" height="1024" alt="Independent PowerDNS Authoritative Primary and two Authoritative Server Secondary" src="https://github.com/user-attachments/assets/c6a99432-ebdf-4554-a893-9a556e93c980" />

It'd be nice if some help.

Thank you,

**Comment by Habbie** [comment]
> I have a problem with the Recursor configuring to work.

What exactly is the problem? Can you show what goes wrong?

**Reply by sysbitnet** [reply]
@Habbie After After a long period of reading, testing, I changed my complete question. The point of this is to help anyone who wants to test in a local lab, school, university, or company to set up basic PowerDNS infrastructure. My goal is to encourage people to try PowerDNS.
